.github/workflows/snap.yaml

```diff
@@ -0,0 +1,65 @@
+name: Snap
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "feature/**"
+  # allow manual re-publishing as branches expire after 30 days
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
```

Comment:
nit: I would either rename this to `publish` or `build and publish`.  Or I would split this into a build step and a publish step.

It's misleading to see a workflow job named "build" that also publishes a snap:

![image](https://github.com/user-attachments/assets/74591a95-0276-48b8-a777-ddc6d046c6f2)